public/components/FeatureAnywhereContextMenu/CreateAnomalyDetector/AssociateExisting/containers/AssociateExisting.tsx

```diff
@@ -0,0 +1,282 @@
+        const savedAugmentObjectsArr: ISavedAugmentVis[] = get(
+          resp,
+          'hits',
+          []
+        );
+        const curDetectorsToDisplayOnList =
+          getExistingDetectorsAvalableToAssociate(
+            Object.values(allDetectors),
+            savedAugmentObjectsArr
+          );
+        setExistingDetectorsAvailableToAssociate(curDetectorsToDisplayOnList);
+        setIsLoadingFinalDetectors(false);
+      }
+    });
+  }, [allDetectors]);
+
+  // cross checks all the detectors that exist with all the savedAugment Objects to only display ones
+  // that are associated to the current visualization
+  const getExistingDetectorsAvalableToAssociate = (
+    detectors: DetectorListItem[],
+    savedAugmentObjects: ISavedAugmentVis[]
+  ) => {
+    // Filter all savedAugmentObjects that aren't linked to the specific visualization
+    const savedAugmentForThisVisualization: ISavedAugmentVis[] =
+      savedAugmentObjects.filter(
+        (savedObj) =>
+          get(savedObj, 'visId', '') === associateExistingProps.embeddableVisId
+      );
+
+    // Map all detector IDs for all the found augmented vis objects
+    const savedAugmentDetectorsSet = new Set(
+      savedAugmentForThisVisualization.map((savedObject) =>
+        get(savedObject, 'pluginResourceId', '')
+      )
+    );
+
+    // detectors here is all detectors
+    // for each detector in all detectors return that detector if that detector ID isnt in the set
+    // filter out any detectors that aren't on the set of detectors IDs from the augmented vis objects.
+    const detectorsToDisplay = detectors.filter((detector) => {
+      if (
+        !savedAugmentDetectorsSet.has(detector.id) &&
+        detector.detectorType === 'SINGLE_ENTITY'
+      ) {
+        return detector;
+      }
+    });
+    return detectorsToDisplay;
+  };
+
+  useEffect(() => {
+    getDetectors();
+  }, []);
+
+  const getDetectors = async () => {
+    dispatch(getDetectorList(GET_ALL_DETECTORS_QUERY_PARAMS));
+  };
+
+  const selectedOptions = useMemo(() => {
+    if (
+      !existingDetectorsAvailableToAssociate ||
+      !associateExistingProps.selectedDetector
+    ) {
+      return [];
+    }
+
+    const detector = (existingDetectorsAvailableToAssociate || []).find(
+      (detector) =>
+        detector.id === get(associateExistingProps.selectedDetector, 'id', '')
+    );
+    return detector ? [{ label: detector.name }] : [];
+  }, [
+    associateExistingProps.selectedDetector,
+    existingDetectorsAvailableToAssociate,
+  ]);
+
+  const detector = useMemo(
+    () =>
+      existingDetectorsAvailableToAssociate &&
+      associateExistingProps.selectedDetector &&
+      existingDetectorsAvailableToAssociate.find(
+        (detector) =>
+          detector.id === get(associateExistingProps.selectedDetector, 'id', '')
+      ),
+    [
+      associateExistingProps.selectedDetector,
+      existingDetectorsAvailableToAssociate,
+    ]
+  );
```

Comment:
not super familiar with `useMemo` here, what is this assigning to `detector`? If we already have the selected detector passed in props, what is the difference?

Reply:
I actually originally was only passing the selectedDetectorId but since I needed the name also in the parent component I changed it to the detector as the prop. The useMemo was just to save the processing time from searching for the correct detector each time but since I know pass the detector itself its not needed.

I will change this